cluster/log-dump/log-dump.sh

```diff
@@ -141,7 +141,7 @@ function save-logs() {
       fi
     else
       case "${KUBERNETES_PROVIDER}" in
-        gce|gke)
+        gce|gke|kubernetes-anywhere)
```

Comment:
As a heads up, a bug I ran into iterating on the cluster log dump logic is a nil pointer reference:
```
	var cmd *exec.Cmd
	// Temporarily set the provider to be gce for the purposes of log dumping.
	cmd.Env = os.Environ()
	cmd.Env = append(cmd.Env, "KUBERNETES_PROVIDER=gce")
	if gcsPath != "" {
		log.Printf("Dumping logs from nodes to GCS directly at path: %v", gcsPath)
		cmd = exec.Command(logDumpPath, localPath, gcsPath)
	} else {
		log.Printf("Dumping logs locally to: %v", localPath)
		cmd = exec.Command(logDumpPath, localPath)
	}
	return finishRunning(cmd)
```

It will need to be fixed with something like this:
```
	var cmd *exec.Cmd
	if gcsPath != "" {
		log.Printf("Dumping logs from nodes to GCS directly at path: %v", gcsPath)
		cmd = exec.Command(logDumpPath, localPath, gcsPath)
	} else {
		log.Printf("Dumping logs locally to: %v", localPath)
		cmd = exec.Command(logDumpPath, localPath)
	}
	// Temporarily set the provider to be gce for the purposes of log dumping.
	cmd.Env = os.Environ()
	cmd.Env = append(cmd.Env, "KUBERNETES_PROVIDER=gce")
	return finishRunning(cmd)
```

https://github.com/kubernetes/test-infra/blob/c24a882fca1afb589a82549d9f0b88f3f6b28697/kubetest/anywhere.go#L272

Reply:
@jessicaochen i am removing the whole back out. PTAL at kubernetes/test-infra#5164